lstm.py

```diff
@@ -0,0 +1,225 @@
+import time
+import numpy as np
+import tensorflow as tf
+from model import LanguageModel
+from utils import DataReader
+
+class Config(object):
+  """
+  Holds model hyperparams and data information.
+  """
+  batch_size = 64
+  state_size = 512
+  embed_dim = 100
+  vocab_size = 20000 + 2 # for UNK TODO
+  sentence_length = 30
+  data_path = "data/sentences.train"
+  learning_rate = 0.5
+  epochs = 1
+
+class Lstm(LanguageModel):
+
+
+  def _get_variable(self, name, shape, weight_decay=None):
+    '''
+    The weight decay parameter gives the scaling constant for the
+    L2-loss on the parameters.
+    '''
+    var = tf.get_variable(name, shape,
+      initializer=tf.contrib.layers.xavier_initializer())
+
+    if weight_decay is not None:
+        wd = tf.mul(tf.nn.l2_loss(var), weight_decay, name='weight_loss')
+        tf.add_to_collection('losses', wd)
+
+    return var
+
+
+  def load_data(self):
+    print "loading data.."
+    data_reader = DataReader()
+    data_reader.construct(self.config.data_path,
+      self.config.vocab_size, self.config.sentence_length)
+    return data_reader
+
+
+  def add_placeholders(self):
+
```

Comment:
nitpick. I know this is probably the default spacing from Stanford but can we all agree that it's stupid and it sucks? I vote - get rid of this extra whitespace.